lib/castle/client.rb

```diff
@@ -6,27 +6,39 @@ class Client
 
     def initialize(request, options = {})
       @do_not_track = default_tracking(options)
-      cookies = options[:cookies] || request.cookies
-      client_id = Extractors::ClientId.new(request, cookies).call('__cid')
+      request_headers = Extractors::Headers.new(request).call
       ip = Extractors::IP.new(request).call
-      headers = Extractors::Headers.new(request).call
-      @api = API.new(client_id, ip, headers)
+      @context = setup_context(request_headers, ip, options[:context])
+      @castle_headers = setup_castle_headers(request, request_headers, ip, options[:cookies])
```

Comment:
sending the `castle_headers` is no longer needed since the data is sent in the context payload